src/nvim_parinfer/main.cljs

```diff
@@ -13,48 +11,48 @@
 
 (defn split-lines [s]
   (when s
-    (string/split s #"\r?\n" -1)))
+    (.split s #"\r?\n")))
 
 (defn run-indent [current-text opts mode]
   (cond
-   (= "indent" mode)
-   (parinfer/indentMode current-text opts)
+    (= "indent" mode)
+    (parinfer/indentMode current-text opts)
 
-   (= "paren" mode)
-   (parinfer/parenMode current-text opts)))
+    (= "paren" mode)
+    (parinfer/parenMode current-text opts)))
 
 (defn format-lines [current-lines cursor-x cursor-line bufnum mode]
   (try
-   (let [opts (clj->js {"cursorX" (dec cursor-x) "cursorLine" (dec cursor-line)})
-         current-text (string/join "\n" current-lines)
-         result (js->clj (run-indent current-text opts mode))
-         new-text (get result "text")]
+    (let [opts #js {"cursorX" (dec cursor-x) "cursorLine" (dec cursor-line)}
+          current-text (.join current-lines "\n")
+          result (run-indent current-text opts mode)
+          new-text (.-text result)]
 
-    (when (not= current-text new-text)
-     (split-lines new-text)))
+      (when (not= current-text new-text)
+        (split-lines new-text)))
 
-   (catch :default e
-     (dbg "EXCEPTION" e e.stack))))
+    (catch :default e
+      (dbg "EXCEPTION" e e.stack))))
 
 (defn parinfer-indent
   [nvim args [[_ cursor-line cursor-x _] bufnum lines mode] nvim-callback]
   (let [start (js/Date.)]
-    (if-let [new-lines (format-lines (js->clj lines) cursor-x cursor-line bufnum mode)]
+    (if-let [new-lines (format-lines lines cursor-x cursor-line bufnum mode)]
       (do
-       #_(js/debug "c" (- (.getTime (js/Date.)) (.getTime start)))
-       (nvim-callback nil (clj->js new-lines)))
+        #_(js/debug "c" (- (.getTime (js/Date.)) (.getTime start)))
+        (nvim-callback nil (clj->js new-lines)))
       (do
-       #_(js/debug "n" (- (.getTime (js/Date.)) (.getTime start)))
-       (nvim-callback nil (clj->js []))))))
+        #_(js/debug "n" (- (.getTime (js/Date.)) (.getTime start)))
+        (nvim-callback nil (clj->js []))))))
 
```

Comment:
Good catch, fixed in bf33d085bfd4ee88a202faa94f07fe3a32eea2a7.

re: #js reader macro, the difference is probably negligible for the empty array and the parinfer opts, but I checked, and it does emit a JS literal rather than a call to clj->js, so it should be a tiny bit faster.